dallinger/recruiters.py

```diff
@@ -128,7 +132,7 @@ def exit_response(self, experiment, participant):
         """
         raise NotImplementedError
 
-    def reward_bonus(self, assignment_id, amount, reason):
+    def reward_bonus(self, worker_id, assignment_id, amount, reason):
```

Comment:
I don't love that we need to make an API/signature change for this. Maybe there's a way we could make the arguments more flexible? I guess that's tricky because this is always called from the worker, would passing the participant object provide a simpler API with more flexibility (though possibly harder to test)?

Reply:
I wrestled with this, too, for the same reasons.  I'm wondering about a `participant_data` parameter, which would just be data (just a `dict` probably) which might get us both benefits. In fact, we already have this in the form of `Participant.json_data()`, except that it mysteriously lacks the DB `id`, but that could be added.

Reply:
I like the idea of using the participant `json_data` as `participant_data` if we're changing the signature. What's the need for the DB `id`? Can't we just look that up using `worker_id` or `assignment_id`?

I'm not sure I understand why we can't just pass the participant object here, since the worker and the recruiter method presumably run in the same transaction/session, but if there's an issue with that, the `json_data` should be relatively future proof, I'd think.

Reply:
I wanted to avoid immediately succumbing to the temptation to pass an ORM object when only data is necessary, but we do pass `Participant` objects to some other methods in the `Recruiter` API, so the precedent is already set, for better or worse. What tips the balance: I just discovered that `worker_id` is not in `json_data` either, probably because this is intended to be safely exportable without PII. We definitely need that for rewarding bonuses!

Instead of adding another `[x]_data` method I'll just relent an pass the `Participant`.